packages/devextreme/js/__internal/ui/calendar/m_calendar.ts

```diff
@@ -250,116 +258,134 @@ class Calendar<
       },
-      tab() {},
+      tab(): void {},
       enter: this._enterKeyHandler,
     };
   }
 
-  _enterKeyHandler(e): void {
+  _enterKeyHandler(e: KeyboardEvent): void {
+    const { currentDate = new Date() } = this.option();
     if (!this._isMaxZoomLevel()) {
       this._navigateDown();
-    } else if (!this._view.isDateDisabled(this.option('currentDate'))) {
-      const value = this._updateTimeComponent(this.option('currentDate'));
+    } else if (!this._view.isDateDisabled(currentDate)) {
+      const value = this._updateTimeComponent(currentDate);
       this._selectionStrategy.selectValue(value, e);
     }
   }
 
-  _getSerializationFormat(optionName) {
-    const value = this.option(optionName || 'value');
+  _getSerializationFormat(optionName = 'value'): string | undefined | null {
+    const value = this.option(optionName);
+    const { dateSerializationFormat } = this.option();
 
-    if (this.option('dateSerializationFormat')) {
-      return this.option('dateSerializationFormat');
+    if (dateSerializationFormat) {
+      return dateSerializationFormat;
     }
 
     if (isNumeric(value)) {
       return 'number';
     }
 
     if (!isString(value) || value === '') {
-      return;
+      return undefined;
     }
 
     return dateSerialization.getDateSerializationFormat(value);
   }
 
-  _convertToDate(value) {
+  _convertToDate(value): Date | null {
     return dateSerialization.deserializeDate(value);
   }
 
-  _dateValue(value, event) {
+  _dateValue(value: Date | Date[], event): void {
     if (event) {
       if (event.type === 'keydown') {
         const cellElement = this._view._getContouredCell().get(0);
         event.target = cellElement;
       }
       this._saveValueChangeEvent(event);
     }
-    this._dateOption('value', value);
+    this._setDateOption('value', value);
   }
 
-  _dateOption(optionName, optionValue) {
-    const isArray = optionName === 'value' && !this._isSingleMode();
-    const value = this.option('value');
-
-    if (arguments.length === 1) {
-      return isArray
-        // @ts-expect-error ts-error
-        ? (value ?? []).map((value) => this._convertToDate(value))
-        : this._convertToDate(this.option(optionName));
-    }
-
+  _setDateOption(
+    optionName: 'value' | 'min' | 'max',
+    optionValue: DateLike | DateLike[],
+  ): void {
+    const isArray = optionName === 'value' && !this._isSingleMode() && Array.isArray(optionValue);
     const serializationFormat = this._getSerializationFormat(optionName);
     const serializedValue = isArray
-      ? optionValue?.map((value) => dateSerialization.serializeDate(value, serializationFormat)) || []
+      ? optionValue?.map((value) => dateSerialization.serializeDate(value, serializationFormat))
+        || []
       : dateSerialization.serializeDate(optionValue, serializationFormat);
 
     this.option(optionName, serializedValue);
   }
 
+  _getDateOption(optionName: 'value' | 'min' | 'max'): Date | Date[] | undefined | null {
```

Comment:
hm, why do we have a new method?

Reply:
It's not really a new one, I've just decided to split `_dateOption` on `_getDateOption` and `_setDateOption`, nothing new is added. I belive it's better that a method which can return either undefined or some Calendar option, which will require to add additional checks